solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -0,0 +1,331 @@
+    ///         is challenged and proven to be malicious, each operator who
+    ///         signed the malicious result is slashed for
+    ///         `maliciousDkgResultSlashingAmount`.
+    uint96 public maliciousDkgResultSlashingAmount;
+
+    /// @notice Percentage of the staking contract malicious behavior
+    ///         notification reward which will be transferred to the notifier
+    ///         reporting about a malicious DKG result. Notifiers are rewarded
+    ///         from a notifiers treasury pool. For example, if
+    ///         notification reward is 1000 and the value of the multiplier is
+    ///         5, the notifier will receive: 5% of 1000 = 50 per each
+    ///         operator affected.
+    uint256 public maliciousDkgResultNotificationRewardMultiplier;
+
+    // External dependencies
+
+    SortitionPool public immutable sortitionPool;
+    /// TODO: Add a dependency to `threshold-network/solidity-contracts` and use
+    /// IStaking interface from there.
+    IWalletStaking public immutable staking;
+
+    // Events
+
+    event DkgStarted(uint256 indexed seed);
+
+    event DkgResultSubmitted(
+        bytes32 indexed resultHash,
+        uint256 indexed seed,
+        DKG.Result result
+    );
+
+    event DkgTimedOut();
+
+    event DkgResultApproved(
+        bytes32 indexed resultHash,
+        address indexed approver
+    );
+
+    event DkgResultChallenged(
+        bytes32 indexed resultHash,
+        address indexed challenger,
+        string reason
+    );
+
+    event DkgStateLocked();
+
+    event DkgSeedTimedOut();
+
+    event WalletCreated(
+        bytes32 indexed walletID,
+        bytes32 indexed dkgResultHash
+    );
+
+    event SignatureRequested(bytes32 indexed walletID, bytes32 indexed digest);
+
+    event SignatureSubmitted(
+        bytes32 indexed walletID,
+        bytes32 indexed digest,
+        Wallets.Signature signature
+    );
+
+    event DkgMaliciousResultSlashed(
+        bytes32 indexed resultHash,
+        uint256 slashingAmount,
+        address maliciousSubmitter
+    );
+
+    event DkgMaliciousResultSlashingFailed(
+        bytes32 indexed resultHash,
+        uint256 slashingAmount,
+        address maliciousSubmitter
+    );
+
+    constructor(
+        SortitionPool _sortitionPool,
+        IWalletStaking _staking,
+        DKGValidator _dkgValidator,
+        address _walletOwner
+    ) {
+        sortitionPool = _sortitionPool;
+        staking = _staking;
+        walletOwner = _walletOwner;
+
+        // TODO: Implement governance for the parameters
+        // TODO: revisit all initial values
+
+        maliciousDkgResultSlashingAmount = 50000e18;
+        maliciousDkgResultNotificationRewardMultiplier = 100;
+
+        dkg.init(_sortitionPool, _dkgValidator);
+        dkg.setResultChallengePeriodLength(11520); // ~48h assuming 15s block time
+        dkg.setResultSubmissionPeriodLength(100 * 20); // TODO: Verify value
```

Comment:
I think it's cheaper if we provide already calculated value 2000. We can put 100 * 20 in a comment.